src/sentry/static/sentry/app/views/usageStats/usageChart/index.tsx

```diff
@@ -0,0 +1,337 @@
+    value: DataCategory.TRANSACTIONS,
+    disabled: false,
+  },
+  {
+    label: 'Attachments',
+    value: DataCategory.ATTACHMENTS,
+    disabled: false,
+  },
+];
+
+export enum ChartDataTransform {
+  CUMULATIVE = 'cumulative',
+  DAILY = 'daily',
+}
+
+export const CHART_OPTIONS_DATA_TRANSFORM: SelectValue<ChartDataTransform>[] = [
+  {
+    label: 'Cumulative',
+    value: ChartDataTransform.CUMULATIVE,
+    disabled: false,
+  },
+  {
+    label: 'Day-to-Day',
+    value: ChartDataTransform.DAILY,
+    disabled: false,
+  },
+];
+
+export enum SeriesTypes {
+  ACCEPTED = 'Accepted',
+  DROPPED = 'Dropped',
+  PROJECTED = 'Projected',
+}
+
+type Props = {
+  theme: Theme;
+
+  title?: React.ReactNode;
+  footer?: React.ReactNode;
+
+  dataCategory: DataCategory;
+  dataTransform: ChartDataTransform;
+
+  /**
+   * If the parent component will handle data rolling up between cumulative
+   * or daily, then UsageChart will skip the transformation
+   */
+  skipDataTransform: boolean;
+
+  usageDateStart: string;
+  usageDateEnd: string;
+  usageStats: ChartStats;
+
+  /**
+   * Additional data to draw on the chart alongside usage
+   */
+  chartSeries?: EChartOption.Series[];
+
+  /**
+   * Replace default tooltip
+   */
+  chartTooltip?: EChartOption.Tooltip;
+};
+
+type State = {
+  xAxisDates: string[];
+};
+
+export type ChartStats = {
+  accepted: NonNullable<EChartOption.SeriesBar['data']>;
+  dropped: NonNullable<EChartOption.SeriesBar['data']>;
+  projected: NonNullable<EChartOption.SeriesBar['data']>;
+};
+
+export class UsageChart extends React.Component<Props, State> {
+  static defaultProps: Partial<Props> = {
+    skipDataTransform: false,
+  };
+
+  state: State = {
+    xAxisDates: [],
+  };
+
+  static getDerivedStateFromProps(nextProps: Readonly<Props>, prevState: State): State {
+    const {usageDateStart, usageDateEnd} = nextProps;
+    const xAxisDates = getDateRange(usageDateStart, usageDateEnd);
+
+    return {
+      ...prevState,
+      xAxisDates,
+    };
+  }
+
+  get chartColors() {
+    const {dataCategory} = this.props;
+
+    if (dataCategory === DataCategory.ERRORS) {
+      return [COLOR_ERRORS, COLOR_ERRORS_DROPPED, COLOR_PROJECTED];
+    }
+
+    if (dataCategory === DataCategory.ATTACHMENTS) {
+      return [COLOR_ATTACHMENTS, COLOR_ATTACHMENTS_DROPPED, COLOR_PROJECTED];
+    }
+
+    return [COLOR_TRANSACTIONS, COLOR_TRANSACTIONS_DROPPED, COLOR_PROJECTED];
+  }
+
+  get chartMetadata() {
+    const {usageStats, dataCategory, dataTransform, skipDataTransform} = this.props;
+    const {xAxisDates} = this.state;
+
+    const display = CHART_OPTIONS_DATACATEGORY.find(o => o.value === dataCategory);
+    if (!display) {
+      throw new Error('Selected item is not supported');
+    }
+
+    const chartData: ChartStats = {
+      accepted: [],
+      dropped: [],
+      projected: [],
+    };
+
+    const isCumulative =
+      !skipDataTransform && dataTransform === ChartDataTransform.CUMULATIVE;
```

Comment:
Would it be more extensible if the data transform was passed as a function? with the total, new value, and dataTransform as parameters? The default prop could do the cumulative / day-day value.

Alternatively should the conversion of stats -> series be something that a parent can replace? It seems like we have similar logic to this already in getsentry.